# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender_api/src/image.rs
# Commit: 4aa1704827b1
# Full Hash: 4aa1704827b131756b1dac86994e6963a2ba144f
# Author: Nicolas Silva <nsilva@mozilla.com>
# Date: 2024-08-08 09:35:37
# Regressor Bug: 1903977
# File Overlap Count: 1
# Description:
#   Bug 1903977 - Add a blob tile pool. r=gfx-reviewers,gw
#   
#   The tile pool keeps a strong reference to all of its allocations and reuses them when their reference count gets down to one.
#   Jemalloc only uses thread-local arenas for small allocations (<496 bytes) so it does not matter what thread the blob tiles which are typically much larger, are deallocated in.
#   
# ==============================================================================

diff -r b6440b259f94 -r 4aa1704827b1 gfx/wr/webrender_api/src/image.rs
--- a/gfx/wr/webrender_api/src/image.rs	Wed Aug 07 17:42:25 2024 +0000
+++ b/gfx/wr/webrender_api/src/image.rs	Wed Aug 07 18:16:58 2024 +0000
@@ -421,7 +421,8 @@
     fn rasterize(
         &mut self,
         requests: &[BlobImageParams],
-        low_priority: bool
+        low_priority: bool,
+        tile_pool: &mut crate::BlobTilePool,
     ) -> Vec<(BlobImageRequest, BlobImageResult)>;
 }
 